### DISCUSSION TITLE
VSCode extension

### DISCUSSION THREAD

**Discussion by Gizzio** [treeverse/dvc#6047] [Show and tell]
It would be great if you develop a VSCode extension allowing manage DVC pipeline in a similar way that it allows managing git. 

---

We are working on VS Code DVC extension for Experiments, Plots and data files. If you'd like to participate in the alpha releases and/or give it a try, please contact `@ivan` on Discord. Check the [announcement](https://discord.com/channels/485586884165107732/497187456051970048/841831363157032960) here.

**Comment by courentin** [comment]
**Idea dropping:**
Something nice to have would be the ability to repro a stage from a `dvc.yaml` file directly (as weel as the ability to launch it in debug mode).

A bit like the "Test" feature of vscode

<img width="711" alt="Screenshot 2021-03-29 at 20 39 33" src="https://user-images.githubusercontent.com/12371832/112883770-fe5b6f80-90ce-11eb-8678-8db74f82c3d2.png">

**Comment by skshetry** [comment]
We are working on VS Code DVC extension for Experiments, Plots and data files. If you'd like to participate in the alpha releases and/or give it a try, please contact `@ivan` on Discord. Please check the [announcement](https://discord.com/channels/485586884165107732/497187456051970048/841831363157032960) here.

**Reply by dynamicwebpaige** [reply]
This is awesome to see, and I would love to help (if I could be useful!). Do you have an invite link to the Discord server? Am not able to view the announcement link in the channel directly.

**Reply by skshetry** [reply]
@dynamicwebpaige, here's the invite: https://dvc.org/chat.

**Reply by shcheklein** [reply]
@dynamicwebpaige that would be great! I've invited you to the project and Figma. Discord - ping me there if you join.

**Comment by mattseddon** [comment]
We aren't finished but we have made a start. The extension is now available for general use:

- https://github.com/iterative/vscode-dvc
- https://marketplace.visualstudio.com/items?itemName=Iterative.dvc
